documentation/adoc/appendix_metrics.adoc

```diff
@@ -0,0 +1,131 @@
+[appendix]
+== Metrics
+
+This section describes how to deploy a Prometheus server for scraping metrics from the Kafka cluster and showing them using a Grafana dashboard. The resources provided are examples to show how Kafka metrics can be stored in Prometheus: They are not a recommended configuration, and further support should be available from the Prometheus and Grafana communities.
+
+When using `minikube` or `minishift` for your Apache Kafka deployment adding Prometheus and Grafana servers, it would be better to increase the default amount of memory available to the virtual machine (i.e. not using the default 2 GB of RAM but 4 GB instead).
+
+Information on how to increase the default amount of memory can be found in the following section <<_installing_kubernetes_and_openshift_cluster>>
+
+=== Deploying on OpenShift
+
+==== Prometheus
+
+The Prometheus server configuration uses a service discovery feature in order to discover the pods in the cluster from which it gets metrics.
+In order to have this feature working, it's necessary to provide access to the API server to get the pod list to the service account used for running the Prometheus service pod (the service account `prometheus-server` is used).
+
+[source]
+export NAMESPACE=[namespace]
+oc login -u system:admin
+oc create sa prometheus-server
+oc adm policy add-cluster-role-to-user cluster-reader system:serviceaccount:${NAMESPACE}:prometheus-server
+oc login -u developer
+
+where `[namespace]` is the namespace/project where the Apache Kafka cluster was deployed.
+
+After that, it's possible to deploy the Prometheus server in the cluster :
+
+. Create the https://github.com/strimzi/strimzi/blob/master/metrics/examples/prometheus/openshift-template.yaml[provided "prometheus" template] by running
++
+[source]
+oc create -f metrics/examples/prometheus/openshift-template.yaml
+
+. Create a new app using the "prometheus" template:
++
+[source]
+oc new-app prometheus
+
+==== Grafana
+
+A Grafana server is necessary only to get a visualisation of the Prometheus metrics. The following https://github.com/OpenShiftDemos/grafana-openshift[project] can be used for visualisation which you need to clone locally.
+
+From the cloned project directory, run
```

Comment:
We have a Grafana image on our Docker Hub as `strimzi/grafana:latest`. This is also used in the `metrics/grafana/kubernetes.yaml`. Why do we here and in the Kubernetes section ask the user to build the image? I think we should just tell him to deploy the grafana image. 

@ppatierno Do you remember why have you wrote it like this originaly? Why do we have the image in Docker Hub and yet build it from scratch here?

Reply:
There seems to be an image grafana/grafana on Docker Hub. I guess we should use that one.

Reply:
Regarding our image on Strimzi, we should give it a try using the Docker Hub image before deleting this sentence in order to be sure that it works; I don't see any reason why the user should build it if the image is in Docker Hub.
Regarding using the "official" grafana/grafana I remember that there are some problems around user permissions in that image and for this reason guys from OpenShift created the project we are using.

Reply:
Oki, if the official image doesn't work (I can imagine some issues with permissions). In any case we might simply build the image and put it on our Docker Hub (or the stirmzi-lab Docker Hub) and use it from there.

@serrss Wanna learn more about the monitoring? Wanna try to run it with OpenShift to see if the Grafana image works?